invertedai/cosimulation.py

```diff
@@ -1,114 +1,75 @@
+    All keyword arguments to :func:`large_initialize` can be passed to the constructor 
+    here and a sufficient combination of them must be passed as required by :func:`large_initialize`.
     This wrapper caches static agent properties and propagates the recurrent state,
     so that only states of ego agents and NPCs need to be exchanged with it to
-    perform co-simulation. Typically, each time step requires a single call to
-    :func:`self.npc_states` and a single call to :func:`self.step`.
+    perform co-simulation. Typically, each time step requires a single call to :func:`self.step`.
 
     This wrapper only supports a minimal co-simulation functionality.
-    For more advanced use cases, call :func:`initialize` and :func:`drive` directly.
-
-    :param location: Location name as expected by :func:`initialize`.
-    :param ego_agent_mask: List indicating which agent is ego, meaning that it is
-        controlled by you externally. The order in this list should be the same as that
-        used in arguments to :func:`initialize`.
-    :param monitor_infraction: Whether to monitor driving infractions, at a small increase
-        in latency and payload size.
-    :param get_birdview: Whether to render the bird's eye view of the simulation state
-        at each time step. It drastically increases the payload received from Inverted AI
-        servers and therefore slows down the simulation - use only for debugging.
-    :param random_seed: Controls the stochastic aspects of simulation for reproducibility.
+    For more advanced use cases, call :func:`large_initialize` and :func:`large_drive` directly.
+
+    :param location: 
+        Location name as expected by :func:`large_initialize` and :func:`large_drive`.
+    :param ego_agent_properties: 
+        Agent properties for all ego agents NOT controlled by the Inverted AI API but nonetheless
+        must be given to the API so the NPC agents are aware of them. Please refer to the documentation
+        for :func:`large_initialize` for more information on how to format this parameter (treating the
+        ego agents as "predefined agents".)
+    :param ego_agent_agent_states: 
+        Agent states for all ego agents NOT controlled by the Inverted AI API but nonetheless
+        must be given to the API so the NPC agents are aware of their states. Please refer to the documentation
+        for :func:`large_initialize` for more information on how to format this parameter (treating the
+        ego agents as "predefined agents".)
     """
 
     def __init__(
         self,
         location: str,
-        ego_agent_mask: Optional[List[bool]] = None,
-        monitor_infractions: bool = False,
-        get_birdview: bool = False,
-        random_seed: Optional[int] = None,
-        traffic_lights: bool = False,
-        # sufficient arguments to initialize must also be included
-        **kwargs,
+        ego_agent_properties: Optional[List[AgentProperties]] = None,
```

Comment:
Are`ego_agent_properties` and `ego_agent_agent_states` bascially the conditional agents information passed down to INITIALIZE ?
Shouldn't `conditional_` be a better prefix than `ego_` here ?

Also i notice the new interface doesn't allow us to use a separate mask to decide which agents to control anymore, instead it is expecting the user to control all the passed ego agents. Would there be a case where they pre-define some agents but only want to control a subset of them ?